main.rb

```diff
@@ -0,0 +1,53 @@
+# this function accepts an array of numbers to be sorted
+# it sorts them and returns the sorted array
+def bubble_sort(arr)
+  sorted = false # a control for the sorting while loop
+
+  while (!sorted) do # keep iterating while sorted is false
+    operate = false
+    arr.length.times do |i|
+      current_number = arr[i]
+      next_number = arr[i + 1] || current_number
+
+      if current_number > next_number # compare the current and next number then swap is current is bigger than next
+        arr[i] = next_number
+        arr[i + 1] = current_number 
+        operate = true
+      end
+    end
+
+    if operate == false
+      sorted = true
+    end
+  end
+
+  puts "#{arr}"
+  return arr
+end
+
+bubble_sort([4,3,78,2,0,2]) # [0,2,2,3,4,78]
+bubble_sort([23, 8, 12, 0, 12, 7, 2, 4, 10, 1]) # [0, 1, 2, 4, 7, 8, 10, 12, 12, 23]
+
+def bubble_sort_by(arr)
+  sorted = false
+
+  while(!sorted) do
+    operation = false
+    arr.length.times do |i|
+      current_element = arr[i]
+      next_element = arr[i + 1] || current_element
+
+      if yield(current_element,next_element) > 0
+        arr[i] = next_element
+        arr[i + 1] = current_element
+        operation = true
+      end
+    end
+    break unless sorted
+  end
+  puts "#{arr}"
+end
+
+bubble_sort_by(["hi","hello","hey"]) do |left,right|
+  left.length - right.length
+end
```

Comment:
#### `REQUIRED`

Add `.stickler.yml` file to the root of the project. Also let's use [this file](https://github.com/microverseinc/linters-config/blob/master/ruby/.stickler.yml). :point_down: 

``` yml
linters:
  rubocop:
    display_cop_names: true
files:
  ignore:
    - "bin/*"
    - "db/*"
    - "config/*"
    - "Guardfile"
    - "Rakefile"
```
- Also, add the `.rubocop.yml` file, this can be found [here](https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml)

> Remember that `.stickler.yml` and `.rubocop.yml` file names start with a **dot**

If you are having an issue with setting up linter, read the `Troubleshooting` section of [this repo](https://github.com/microverseinc/linters-config/tree/master/ruby#troubleshooting).

### You can run the linter locally:  💥 - run `rubocop` in your local env:
>     - add `gem 'rubocop'` to `Gemfile` (not sure how to use Gemfile? Read [this](https://bundler.io/v1.15/guides/bundler_setup.html))
>     - run `bundle install`
>     - copy [.rubocop.yml](https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml) to the root directory of your project
>     - run `rubocop`
>     - fix linter errors
>     - **Remember to let your Code Reviewer know that you had problems with Stickler and you used linter in local env.**

Note: You can autofix comment style rules with `rubocop -a`

Reply:
this is an insightful review
Thank you!

Reply:
Hey @jalasem this project has been approved. I submitted the link of my repo sorry. You can view it here. https://github.com/damiecode/bubble_sort-rb/pull/1